.template/lib/utils/wrappers/permission_wrapper.dart

```diff
@@ -0,0 +1,31 @@
+import 'package:injectable/injectable.dart';
+import 'package:permission_handler/permission_handler.dart'
+    as permission_handler;
+
+abstract class PermissionWrapper {
+  Future<bool> requestCameraPermission();
+
+  Future<bool> isCameraPermissionDenied();
+
+  Future<bool> isCameraPermissionPermanentlyDenied();
```

Comment:
@luongvo any idea on how are we going to initiate a project with or without these classes? Let's say in a project where we don't have a need to use them.

Reply:
@sleepylee Actually we should have a base code for future implementations. How about commenting all of them as a guideline for the developer to start their first implementation, the same as what we did with [iOS permission config](https://github.com/nimblehq/flutter_templates/pull/109/files#diff-029fdb263ed8cd70d21ed721be6b9c9741c0ee03384c4200c7a89735285760b4)? cc @doannimble  @markgravity

Reply:
This new [wiki documenting ticket](https://github.com/nimblehq/flutter_templates/issues/110) somehow solves your concern a bit as well @sleepylee 💪

Reply:
I believe what Manh did before was to have pluggable code where you can enable them from the init command. I would prefer to not have any concrete implementations and later requiring the devs to remove them manually. Is there anything you can design this feature?

Reply:
Sorry for late 🙇 
@luongvo I think we should not add the concrete implementation into flutter_template. Otherwise, the dev may have to remove it manually if they don't need it, and I think you know that pain, right? 😓 

I think we should have the pluggable code tool to help us add/ remove the features when initializing the project.

Previously, I found [mason tool](https://nimble-co.slack.com/archives/C01NKCFNE6R/p1656488939246219) that could match our needs, you should give it a try 😉 What do you think?

Reply:
@manh-t While waiting for a template generating tool like `mason`, do you agree to have this part in our template in the meantime? Ideally, we can move this part into a configuration after adding that tool 🤔

Reply:
@luongvo yes, we can move this part into mason brick later 👌